client/src/components/Login/PasswordStrength.vue

```diff
@@ -0,0 +1,179 @@
+        passwordStrength.value = "weak";
+    } else if (strengthScore.value <= 3) {
+        passwordStrength.value = "medium";
+    } else {
+        passwordStrength.value = "strong";
+    }
+}
+
+watch(
+    () => props.password,
+    (newPassword) => {
+        if (typeof newPassword === "string") {
+            evaluatePasswordStrength(newPassword);
+        }
+    }
+);
+</script>
+
+<template>
+    <div>
+        <div class="password-strength-bar-container mt-2">
+            <div
+                class="password-strength-bar"
+                :class="passwordStrength"
+                :style="{ width: `${(strengthScore / 4) * 100}%` }"></div>
+        </div>
+
+        <div class="password-info-container mt-2">
+            <div :class="['password-strength', passwordStrength]">
+                <span v-if="passwordStrength === 'empty'"></span>
+                <span v-else-if="passwordStrength === 'weak'">Weak Password</span>
+                <span v-else-if="passwordStrength === 'medium'">Medium Password</span>
+                <span v-else>Strong Password</span>
+            </div>
+            <div v-if="passwordStrength !== 'empty'" class="crack-time">
+                <strong>Estimated time to crack:</strong>
+                <span :class="passwordStrength"> {{ crackTime }}</span>
+            </div>
+        </div>
+
+        <BButton variant="secondary" class="ui-link mt-3" @click="showPasswordGuidelines = true">
+            Password Guidelines
+        </BButton>
+
+        <div>
+            <BModal v-model="showPasswordGuidelines" title="Tips for a secure Password">
+                <p>A good password should meet the following criteria:</p>
+                <ul>
+                    <li>At least 11 characters long.</li>
+                    <li>Use uppercase and lowercase letters.</li>
+                    <li>At least one number and one special character.</li>
+                    <li>Avoid common passwords like <code>123456</code> or <code>password</code>.</li>
+                    <li>No repeated patterns like <code>aaaa</code> or <code>123123</code>.</li>
+                </ul>
+                <p>
+                    Learn more about:
+                    <a
+                        href="https://www.cisa.gov/secure-our-world/use-strong-passwords target="
+                        target="_blank"
+                        rel="noopener noreferrer"
+                        >strong passwords</a
+                    >.
+                </p>
+                <template v-slot:modal-footer>
+                    <BButton variant="secondary" @click="showPasswordGuidelines = false">Schließen</BButton>
+                </template>
+            </BModal>
+        </div>
+    </div>
+</template>
+
+<style scoped lang="scss">
+@import "theme/blue.scss";
+
+.password-strength-bar-container {
+    background-color: $gray-200;
+    height: 8px;
+    border-radius: 4px;
+    overflow: hidden;
+    margin-top: 5px;
```

Comment:
I recommend using Bootstrap utility classes instead of pixel and rem values. For example, use mt-2 directly in the HTML class attribute or apply it in CSS with @extend .mt-2;.

Reply:
We really need to have a talk about bootstrap and bootstrap utility classes, since they are hurting our code quality and maintainability. I've discussed this with some people already, and am currently working on an outline of why and how we can proceed with this issue with my findings. I will post this under discussions once ready, and would ask you to wait for that, since this review is not the right place for this discussion.